src/data.js

```diff
@@ -31,14 +52,7 @@ function dataAttr( elem, key, data ) {
 
 		if ( typeof data === "string" ) {
 			try {
-				data = data === "true" ? true :
-					data === "false" ? false :
-					data === "null" ? null :
-
-					// Only convert to a number if it doesn't change the string
-					+data + "" === data ? +data :
-					rbrace.test( data ) ? JSON.parse( data ) :
-					data;
+				data = getData( data );
```

Comment:
We should consider disabling `no-nested-ternary`... this is getting too aggressive for too little benefit (and even negative benefit, though I know which side of that argument I'm on).